### DISCUSSION TITLE
who set  command timeout in web.config ?

### DISCUSSION THREAD

**Discussion by MajidSafari** [linq2db/linq2db#3063] [Q&A] [answered]
hi, 
my connectionstring  in web.config.
 i want set timeout in connectionstring but , linq2db not read !!

how set timeout  in web.config ??

tanx , bro

**Comment by MaceWindu** [comment, marked as answer]
Connection string used only by database provider. Not all providers support command timeout in connection string (e.g. sql server providers doesn't have such option for connection string).

You should check connection string documenation for your provider/database, which you didn't specified.

**Comment by MajidSafari** [comment]
@MaceWindu tanx

if you can add property in T4 add CommandTimeout  ,

`		public db(string configuration)
			: base(configuration)
		{
			InitDataContext();
			InitMappingSchema();
			CommandTimeout = 10;
		}`

**Comment by viceroypenguin** [comment]
@MajidSafari  - you can create your own constructor to do this, since the generated class is a partial class. Update your .tt file to add `GenerateConstructors = false;` to turn off generating the constructors automatically, and put your custom constructor in a separate file.

**Comment by MajidSafari** [comment]
Hi @viceroypenguin , Tanx

i try ... but icant :(

**Comment by viceroypenguin** [comment]
What difficulties are you having in doing this?
